lib/templates.js

```diff
@@ -1,81 +1,110 @@
+const _ = require('lodash');
 
 module.exports = function(client) {
-  var templates = {
-    all: function(callback) {
+  return {
+    /**
+     * List an overview of all templates.
+     *
+     * @param {RequestCb} [callback]
+     * @returns {Promise}
+     */
+    list: function(callback) {
       var options = {
         uri: api
       };
       return client.get(options).asCallback(callback);
     },
-    find: function(options, callback) {
+    /**
+     * Get details about a specified template by its id.
+     *
+     * @param {string} id
+     * @param {Object} options
+     * @param {RequestCb} [callback]
+     * @returns {Promise}
+     */
+    get: function(id, options, callback) {
       var reqOpts;
       options = options || {};
 
-      if(!options.id) {
+      if(!id) {
         return Promise.reject(new Error('template id is required')).asCallback(callback);
       }
 
       reqOpts = {
-        uri: api + '/' + options.id
+        uri: api + '/' + id
+        , qs: options
       };
 
-      if(options.draft) {
-        reqOpts.qs = reqOpts.qs || {};
-        reqOpts.qs.draft = options.draft;
-      }
-
       return client.get(reqOpts).asCallback(callback);
     },
-    create: function(options, callback) {
+    /**
+     * Create a new template.
+     *
+     * @param {Object} template
+     * @param {RequestCb} [callback]
+     * @returns {Promise}
+     */
+    create: function(template, callback) {
       var reqOpts;
-      options = options || {};
 
-      if (!options.template) {
+      if (!template || typeof template !== 'object') {
         return Promise.reject(new Error('template object is required')).asCallback(callback);
       }
 
       reqOpts = {
         uri: api
-        , json: toApiFormat(options.template)
+        , json: template
       };
 
       return client.post(reqOpts).asCallback(callback);
     },
-    update: function(options, callback) {
-      var object, reqOpts;
-      options = options || {};
+    /**
+     * Update an existing template.
+     *
+     * @param {string} id
+     * @param {Object} template
+     * @param {RequestCb} callback
+     * @returns {Promise}
+     */
+    update: function(id, template, callback) {
+      var reqOpts;
 
-      if(!options.template) {
+      if(!id) {
+        return Promise.reject(new Error('template id is required')).asCallback(callback);
+      }
+
+      if(!template || typeof template !== 'object') {
         return Promise.reject(new Error('template object is required')).asCallback(callback);
       }
 
-      object = toApiFormat(options.template);
       reqOpts = {
-        uri: api + '/' + object.id
-        , json: object
+        uri: api + '/' + id
+        , json: _.cloneDeep(template)
+        , qs: {}
       };
 
-      if(options.update_published) {
-        reqOpts.qs = reqOpts.qs || {};
-        reqOpts.qs.update_published = options.update_published;
+      if(reqOpts.json.update_published) {
```

Comment:
I might add an `options` argument even though its just for 1 field.  We could add it later without breaking the interface to `template.update()` but I like that the object being passed in basically represents the conceptual template object managed through REST.